packages/xo-web/src/common/xo/install-pool-patches-modal/index.js

```diff
@@ -0,0 +1,124 @@
+import _ from 'intl'
+import ActionButton from 'action-button'
+import Component from 'base-component'
+import Icon from 'icon'
+import React from 'react'
+import renderXoItem from 'render-xo-item'
+import SingleLineRow from 'single-line-row'
+import Tooltip from 'tooltip'
+import { Container, Col } from 'grid'
+import { connectStore } from 'utils'
+import {
+  createCounter,
+  createFilter,
+  createGetObject,
+  createGetObjectsOfType,
+  createSelector,
+} from 'selectors'
+import { map, some } from 'lodash'
+import { SelectSr } from 'select-objects'
+import { ejectCd, isSrWritable, setDefaultSr } from 'xo'
+
+@connectStore(() => {
+  const getPool = createGetObject((_, props) => props.pool)
+  return {
+    pool: getPool,
+    poolMaster: createGetObject(
+      createSelector(getPool, ({ master }) => master)
+    ),
+    vms: createGetObjectsOfType('VM'),
+    vbds: createGetObjectsOfType('VBD'),
+  }
+})
+export default class InstallPoolPatchesModalBody extends Component {
+  _getVmsWithCds = createFilter(
```

Comment:
```JS 
_getVmsWitchCdsId = createSelector(
  () => this.props.vbds,
  vbds =>
  map(vbds, vbd => vbd.is_cd_drive && vbd.VDI !== undefined ? vbd.VM : undefined)
    .filter(id => id !== undefined)
)

Reply:
I'm not sure this is the best option. If we take a look at what happens on the first cycle (it's a modal so it's the most important one), it runs 1 big loop (`createFilter`) with 1 small loop inside (`_.some`) while your implementation would run 3 big loops (filter VBDs by pool, `_.map` and `.filter`). Do you agree?
On the other hand, I agree that we can get rid of the VM objects since we can work without them here.